node/caching-compiler.js

```diff
@@ -84,15 +86,27 @@ function readWithCache(info, content) {
   info.cache[cacheFilename] = content;
 
   if (typeof info.cacheDir === "string") {
-    // Writing cache files is something that should only happen at package
-    // development time, so it's acceptable to use fs.writeFileSync
-    // instead of some complicated asynchronous-but-atomic strategy.
-    fs.writeFileSync(absCachePath, content, "utf8");
+    scheduleWrite(absCachePath, content);
   }
 
   return content;
 }
 
+const pendingWrites = Object.create(null);
+let pendingWriteTimer;
+
+function scheduleWrite(path, content) {
+  pendingWrites[path] = content;
+  pendingWriteTimer = pendingWriteTimer || setTimeout(function () {
+    pendingWriteTimer = null;
+    Object.keys(pendingWrites).forEach(function (path) {
+      const content = pendingWrites[path];
+      delete pendingWrites[path];
+      fs.writeFileSync(path, content, "utf8");
+    });
+  }, 10);
+}
```

Comment:
When I first saw the PR title I thought each write would be async but this is a batch of them.
Are there cases where this could get into a cache jam? Could it be writing to a cache file that was written at some point before or could an older cache get written trumping a newer one (some kind of race condition).